### ISSUE TITLE
Typescript errors on `domains` and `customEndpoint`

### ISSUE THREAD

**Issue by genu** [ijkml/nuxt-umami#48]
I'm not sure if this is related to latest Nuxt version, but build a bundle throws typescript errors: (`nuxt build`)

The only configuration I have set are the required env vars:

```bash
NUXT_PUBLIC_UMAMI_HOST="..."
NUXT_PUBLIC_UMAMI_ID="..."
```

<img width="955" alt="CleanShot 2023-04-13 at 01 12 25@2x" src="https://user-images.githubusercontent.com/928780/231660171-23b14f4d-be17-4300-9df4-1ebe53bb7fad.png">

And the relevant code:
<img width="613" alt="CleanShot 2023-04-13 at 01 12 47@2x" src="https://user-images.githubusercontent.com/928780/231660192-ae45a6ad-49e1-452e-82bc-47e79143a937.png">

**Comment by ijkml** [comment]
Hi @genu. I can't seem to reproduce the error. `nr build` works before and after upgrading to Nuxt `v3.4.0`.
Could be related to upgrading. Please run 
```sh
nuxi upgrade --force
```
and let me know if the issue persists.

**Comment by genu** [comment]
I did some more testing and I wasn't able to reproduce it on a new project with only `nuxt-umami` layer, but when I add the `nuxt-umami` layer to an existing project, especially, one that has already a layer the issue shows up.

I'm not quite sure what to make of it.

Nuxt layers have been a hit or miss in my experience to it could be related to Layers not properly working well.

**Comment by genu** [comment]
Another note, I do see that there is intelisense offered if I add the `appConfig` in the `nuxt.config.ts`:
<img width="676" alt="CleanShot 2023-04-13 at 10 46 44@2x" src="https://user-images.githubusercontent.com/928780/231797442-8727f2d6-883f-4022-85de-e1587d71f44e.png">

However, as seen in the screenshot, neither `domains` or `customEndpoint` are included in the intelisense, which are the exact ones that are complained about in the original error.

**Comment by genu** [comment]
I just started using this module, and I thought it was related to the newer Nuxt version, but I don't think that's the case. I updated the title to reflect that.

I'm trying to get a repro together, but its kind of tricky because a basic app doesn't have this issue.

**Comment by genu** [comment]
In the `app.config.d.ts` that is generated, I see that it produces:

```ts
declare const inlineConfig = {
  "umami": {
    "id": "c18ff332-19f0-4fba-bda4-2e2b6d45396b",
    "host": "https://a.binarynights.dev",
    "autoTrack": true,
    "ignoreDnt": true,
    "ignoreLocalhost": false,
    "version": 1
  }
}
```

Is this correct? Shouldn't it include all of the configurations available?

**Comment by ijkml** [comment]
Hmm.
![Screenshot from 2023-04-13 15-50-45](https://user-images.githubusercontent.com/52390439/231798785-e159904d-bf0a-4c57-8759-7b9706e385f8.png)

If a new project doesn't have this issue, how sure are you that it's a Typescript error?

**Comment by ijkml** [comment]
> I'm trying to get a repro together,

Thank you.

**Comment by genu** [comment]
I Created a repo here: https://github.com/genu/umami-repro

NOTE: I wasn't able to reproduce the errors at build time, but on my end I'm able to reproduce the missing `domains` and `customEndpoint` properties from intelisense.

![CleanShot 2023-04-13 at 11 11 51](https://user-images.githubusercontent.com/928780/231804958-af2d14e0-3b35-4568-9b3e-8bc68286f6ba.png)

Also, as an observation:
https://github.com/ijkml/nuxt-umami/blob/b1b8b1a2ee783a42bf2915c8176f23e3b3402d56/nuxt.config.ts#L1-L13

`domain` and `customEndpoint` are the only ones that are initialized to `undefined` could that be causing all of the issues?

**Comment by ijkml** [comment]
Thanks for the repro. Noticed the intellisense issue, too. But build works. 😓

Yes, keen observation, you're right. Would try to test if setting a different value works. I think it's possibly an upstream regression caused by Nuxt 3.4 as things work fine with Nuxt 3.3.

**Comment by genu** [comment]
> Thanks for the repro. Noticed the intellisense issue, too. But build works. 😓
> 
> Yes, keen observation, you're right. Would try to test if setting a different value works. I think it's possibly an upstream regression caused by Nuxt 3.4 as things work fine with Nuxt 3.3.

I'm thinking, in general, initializing values to `undefined` may not be the best practice. Instead of `undefined` maybe initialize `domains` to `[]` and `customEndpoint` to `""`

They are already optional, thus could already be `undefined`. Initializing them as such doesn't quite make sense.

**Comment by ijkml** [comment]
Like I said, this was not an issue pre-3.4 so I don't really know how that does or doesn't make sense.
But, thanks for your input, I'll put that in mind 🙂.

**Comment by ijkml** [comment]
Should now be fixed by `v2.1.4`.
